### PR TITLE
[ios, macos] Fix keypath expressions, json object is parsed incorrectly for nested keypaths.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Fixes https://github.com/mapbox/mapbox-gl-native/issues/11874

### CODE REVIEW

File: platform/darwin/src/NSExpression+MGLAdditions.mm [hunk 1]
```diff
@@ -1069,12 +1069,13 @@ - (id)mgl_jsonExpressionObject {
         }
             
         case NSKeyPathExpressionType: {
-            NSArray *expressionObject;
-            for (NSString *pathComponent in self.keyPath.pathComponents.reverseObjectEnumerator) {
+            NSMutableArray *expressionObject;
+            NSArray *keyPath = [self.keyPath componentsSeparatedByString:@"."];
```
Comment (1ec5) [inline platform/darwin/src/NSExpression+MGLAdditions.mm]: Ooh, good catch – I was probably thinking of `NSString.stringByDeletingPathExtension` here.

File: platform/darwin/src/NSExpression+MGLAdditions.mm [hunk 2]
```diff
@@ -1069,12 +1069,13 @@ - (id)mgl_jsonExpressionObject {
...
+            for (NSString *pathComponent in keyPath.reverseObjectEnumerator) {
                 if (expressionObject) {
-                    expressionObject = @[@"get", pathComponent, expressionObject];
+                    [expressionObject addObject:@[@"get", pathComponent]];
```
Comment (1ec5) [inline platform/darwin/src/NSExpression+MGLAdditions.mm]: `map.box.gl` should become `["get", "gl", ["get", "box", ["get", "map"]]]`, not `["get", "gl", ["get", "box"], ["get", "map"]]`.

Comment (fabian-guerra) [reply platform/darwin/src/NSExpression+MGLAdditions.mm]: That code already handles that case and returns `["get", "gl", ["get", "box", ["get", "map"]]]` I added a test case for that:
https://github.com/mapbox/mapbox-gl-native/pull/11959/files#diff-a4216b00e3cfddef98fc22cbc71f3116R316

File: platform/ios/CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 
 * Unknown tokens in URLs are now preserved, rather than replaced with an empty string. ([#11787](https://github.com/mapbox/mapbox-gl-native/issues/11787))
 * Adjusted when and how the camera transition update and finish callbacks are called, fixing recursion bugs. ([#11614](https://github.com/mapbox/mapbox-gl-native/pull/11614))
+* Fixed an issue preventing nested `keyPath` expressions get parsed accordingly to the spec. ([#11959](https://github.com/mapbox/mapbox-gl-native/pull/11959))
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: Nit: replace `keyPath` with “key path” so developers don’t think this issue was limited to the `+[NSExpression expressionForKeyPath:]` initializer.

**Comment by friedbunny** [comment]
Added the `ios-v4.1.0` milestone — this change is available starting in `ios-v4.1.0-beta.1`.

**Pull request merged**